### DISCUSSION TITLE
Beatmap Listing layout; nx1 instead of nx3 option

### DISCUSSION THREAD

**Discussion by BDSuther** [ppy/osu#28314] [Ideas]
I find it much easier to preview and download multiple maps at a time if I am able to preview the song and then download with minimal mouse movement. The previous layout of 2 items per row was alright but still not ideal. With the update to 3 columns it kind of triggered the "I don't like this" thought. Right now there's only an option to switch between detailed and compact tiles but I am unable to change the number of items per row.

 The classic client's osu!direct UI list was significantly better for this use case as it required one click to preview then another to download. I also liked being able to click anywhere on the list item to get the preview and could mouse over the difficulties while listening and deciding to download or not. When downloading many items after not playing for a while it was easy to just only move the mouse in a small x co-ordinate range and just go up and down the list. With the 3 items per row I will always, by necessity, need to move my mouse around the entire screen to preview and potentially download all the maps.

I would like to know if you think it could be worth allowing a slightly more configurable experience for the beatmap download UI so that  the number of items per row could be changed between [1, 2, 3] or emulate the classic client's osu!direct layout.

One issue I can see with a single item per row with the current implementation is that the preview from the BG image is on the left while the download button is on the right side of the tile. This would basically make the change irrelevant as it would cause even more mouse movement, so it would need some update to the layout of the controls so that less mouse movement to preview and download would be necessary.

**Comment by peppy** [comment]
I can agree that there should be a list view mode.
